marlowe-dashboard-client/src/MainFrame/View.purs

```diff
@@ -1,20 +1,135 @@
 import Data.Maybe (Maybe(..))
 import Effect.Aff.Class (class MonadAff)
-import Halogen (ComponentHTML)
-import Halogen.HTML (button, div, text)
+import Halogen (ComponentHTML, ClassName(ClassName))
+import Halogen.HTML (a, button, div, div_, header, main, nav, span, text)
 import Halogen.HTML.Events (onClick)
-import MainFrame.Types (Action(..), ChildSlots, State)
+import Halogen.HTML.Properties (classes)
+import MainFrame.Types (Action(..), Card(..), ChildSlots, Screen(..), State, Overlay(..))
+import Material.Icons as Icon
+import Notifications.View (renderNotifications)
+import Template.View (renderTemplateLibrary, renderTemplateDetails)
+import Welcome.View (renderHome, renderContracts)
 
-render ::
-  forall m.
-  MonadAff m =>
-  State ->
-  ComponentHTML Action ChildSlots m
+render :: forall m. MonadAff m => State -> ComponentHTML Action ChildSlots m
 render state =
-  div []
-    [ button [ onClick $ const $ Just ClickedButton ] [ text "Click Me" ]
-    , text $ if state.on then "On" else "Off"
+  div
+    [ classes $ ClassName <$> [ "grid", "grid-rows-main", "h-full", "overflow-hidden" ] ]
+    [ header
+        [ classes $ ClassName <$> [ "relative" ] ]
+        [ navbar state.overlay
+        , menu state.overlay state.screen
+        ]
+    , main
+        [ classes $ ClassName <$> [ "relative", "bg-lightblue", "text-blue" ] ]
+        [ screen state
+        , renderNotifications state.overlay state.notifications
+        , card state
+        ]
+    ]
+
+navbar :: forall m. MonadAff m => Maybe Overlay -> ComponentHTML Action ChildSlots m
+navbar currentOverlay =
+  nav
+    [ classes $ ClassName <$> [ "sticky", "top-0", "flex" ] ]
+    [ button
+        [ classes $ ClassName <$> [ "btn", "text-midblue" ] ]
+        [ Icon.image ]
+    , nav
+        [ classes $ ClassName <$> [ "flex", "flex-1", "justify-end" ] ]
+        [ navButton Icon.notifications $ Just $ ToggleOverlay Notifications
+        , navButton menuIcon $ Just $ ToggleOverlay Menu
+        ]
+    ]
+  where
+  menuIcon = if currentOverlay == Just Menu then Icon.close else Icon.menu
+
+  navButton icon action =
+    button
+      [ classes $ ClassName <$> [ "btn", "text-green" ]
+      , onClick $ const $ action
+      ]
+      [ icon ]
+
+menu :: forall m. MonadAff m => Maybe Overlay -> Screen -> ComponentHTML Action ChildSlots m
+menu currentOverlay currentScreen =
+  div
+    [ classes wrapperClasses ]
+    [ nav
+        [ classes $ ClassName <$> [ "bg-gray", "mx-0.5", "flex", "flex-col" ] ]
+        [ menuItem "Home" (Just $ SetScreen Home) (currentScreen == Home)
+        , menuItem "New Contract" (Just $ ToggleCard TemplateLibrary) false
+        , menuItem "Contracts" (Just $ SetScreen Contracts) (currentScreen == Contracts)
+        , menuItem "Docs" Nothing false
+        , menuItem "Support" Nothing false
+        ]
+    ]
+  where
+  wrapperClasses = ClassName <$> [ "absolute", "w-full", "z-20" ] <> if currentOverlay == Just Menu then [] else [ "hidden" ]
```

Comment:
I feel like this pattern will be common and we should have `hideWhen p = if p then ["hidden"] else []` somewhere

Reply:
👍